streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -213,92 +201,93 @@ public boolean process() {
 
         return true;
     }
 
-    private void updateProcessorContext(final ProcessorRecordContext recordContext, final ProcessorNode currNode) {
-        processorContext.setRecordContext(recordContext);
-        processorContext.setCurrentNode(currNode);
-    }
-
-    public boolean requiresPoll() {
-        return requiresPoll;
-    }
-
-    /**
-     * Possibly trigger registered punctuation functions if
-     * current partition group timestamp has reached the defined stamp
-     */
-    public boolean maybePunctuate() {
-        long timestamp = partitionGroup.timestamp();
-
-        // if the timestamp is not known yet, meaning there is not enough data accumulated
-        // to reason stream partition time, then skip.
-        if (timestamp == TimestampTracker.NOT_KNOWN)
-            return false;
-        else
-            return punctuationQueue.mayPunctuate(timestamp, this);
-    }
-
     /**
      * @throws IllegalStateException if the current node is not null
      */
     @Override
-    public void punctuate(ProcessorNode node, long timestamp) {
-        if (processorContext.currentNode() != null)
+    public void punctuate(final ProcessorNode node, final long timestamp) {
+        if (processorContext.currentNode() != null) {
             throw new IllegalStateException(String.format("%s Current node is not null", logPrefix));
+        }
 
-        final StampedRecord stampedRecord = new StampedRecord(DUMMY_RECORD, timestamp);
-        updateProcessorContext(createRecordContext(stampedRecord), node);
+        updateProcessorContext(new StampedRecord(DUMMY_RECORD, timestamp), node);
 
         log.trace("{} Punctuating processor {} with timestamp {}", logPrefix, node.name(), timestamp);
 
         try {
             node.punctuate(timestamp);
-        } catch (KafkaException ke) {
-            throw new StreamsException(String.format("Exception caught in punctuate. taskId=%s processor=%s", id,  node.name()), ke);
+        } catch (final KafkaException e) {
+            throw new StreamsException(String.format("Exception caught in punctuate. taskId=%s processor=%s", id,  node.name()), e);
         } finally {
             processorContext.setCurrentNode(null);
         }
     }
 
+    private void updateProcessorContext(final StampedRecord record, final ProcessorNode currNode) {
+        processorContext.setRecordContext(new ProcessorRecordContext(record.timestamp, record.offset(), record.partition(), record.topic()));
+        processorContext.setCurrentNode(currNode);
+    }
+
     /**
-     * Commit the current task state
+     * <pre>
+     *  - flush state and producer
+     *  - write checkpoint
+     *  - commit offsets
+     * </pre>
      */
+    @Override
     public void commit() {
-        metrics.metrics.measureLatencyNs(time, commitDelegate, metrics.taskCommitTimeSensor);
+        log.trace("{} Committing", logPrefix);
+        metrics.metrics.measureLatencyNs(
+            time,
+            new Runnable() {
```

Comment:
The class is only used once -- and this change keep code local -- when reading code before, it was unclear what's actually happening without stepping into the class that is on a completely different place in the file. Now you can read it at one glance.

I guess it's a personal thing and I revert if you don't like it.